### PR TITLE
[GLUTEN-1956][CH] Fix error conversion of Float32 in CHColumnToSparkRow

### PR DESCRIPTION
**Pull request by exmy**
## What changes were proposed in this pull request?

(Fixes: \#1956)

## How was this patch tested?

UT

### CODE REVIEW

**Comment by exmy** [comment]
cc @taiyang-li @zzcclp

File: cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp [hunk 1]
```diff
@@ -680,10 +680,19 @@ int64_t VariableLengthDataWriter::writeArray(size_t row_idx, const DB::Array & a
             if (elem.isNull())
                 bitSet(buffer_address + offset + start + 8, i);
             else
-                //                 writer.write(elem, buffer_address + offset + start + 8 + len_null_bitmap + i * elem_size);
-                writer.unsafeWrite(
-                    reinterpret_cast<const char *>(&elem.get<char>()),
-                    buffer_address + offset + start + 8 + len_null_bitmap + i * elem_size);
+                // writer.write(elem, buffer_address + offset + start + 8 + len_null_bitmap + i * elem_size);
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp:683]: remove useless code

**Comment by taiyang-li** [comment]
LGTM if remove above useless code

File: cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp (hunk 1 above)
Comment (exmy) [reply cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp]: done

**Review by taiyang-li — APPROVED** [review]

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Pull request merged**